src/mongocxx/test/collection.cpp

```diff
@@ -2406,13 +2406,13 @@ TEST_CASE("create_index tests", "[collection]") {
 
         options::index options{};
         options.unique(true);
+        if (test_util::newer_than(mongodb_client, "4.4"))
```

Comment:
Wrapping the `hidden` option with this check (and below on line 2426) fixed the test failures.